pkg/serializer/func_queue.go

```diff
@@ -3,100 +3,52 @@
 
 package serializer
 
-import (
-	"context"
-	"fmt"
-)
-
-var (
-	// NoRetry always returns false independently of the number of retries.
-	NoRetry = func(int) bool { return false }
```

Comment:
If `NoRetry` is unused, please consider putting the deletion into a separate commit. Combining it into the same commit as the other logic here makes it harder for reviewers to focus on the impacts of individual changes.

Reply:
Not sure what you mean. `NoRetry` was the only value that was ever passed as a `waitFunc` to `Enqueue`, implying it is the default. And since it never retries, that prompted me to remove the retry code. It was never unused, just redundant, and it should remain part of the commit that removed `waitFunc`/`WaitFunc`.

Reply:
Whenever I see a commit that is double-barreled like `datapath: remove unused serializer retry logic, return errors from fc.Wait`, this is an indication to me that there are two logically separate changes that do not need to be in the same commit. Putting them together means that the reviewer needs to track which parts of the commit relate to the first logical change and which relate to the second logical change, and try to identify any changes that don't belong to either (which is often a sign of unintentional changes, which could be a bug). It's just a bit more cognitive work for the reviewer.

The `Wait()`/`Stop()` changes didn't seem to be related to the `NoRetry` change, so while reviewing I went through the above thought process and figured it would have been simpler to understand if `NoRetry` was separate. Similarly if it turns out we miss something, it would be easier to understand where the bug is if we can bisect to one patch with fewer changes.

Reply:
Sure, I'll try to rejig this.